File: Strings/Easy/771. Jewels and Stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {

        int counter = 0;
      
        // for(int i = 0; i < jewels.length(); i++){
        //     for(int j = 0; j < stones.length(); j++){
        //         if(stones.charAt(j) == jewels.charAt(i)){
        //             counter++;
        //         }
        //     }
        // }



        // ArrayList<Character> jewel = new ArrayList<>();
        // for(int i = 0; i < jewels.length(); i++){
        //     jewel.add(jewels.charAt(i));
        // }

        // for(int i = 0; i < stones.length(); i++){
        //     if(jewel.contains(stones.charAt(i))){
        //         counter++;
        //     }
        // }


        for(int i = 0; i < stones.length(); i++){
            if(jewels.indexOf(stones.charAt(i)) != -1){
                counter++;
            }
        }

        return counter;
    }
}

File: Hash Tables/Easy/C++/771. Jewels and Stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        set<char> s;
        int count = 0;

        for(int i = 0; i < jewels.length(); i++){
            s.insert(jewels.at(i));
        }

        for(int i = 0; i < stones.length(); i++){
            if(s.contains(stones.at(i))){
                count++;
            }
        }

        return count;
    }
};
